src/main.c

```diff
@@ -1,30 +1,61 @@
 #include <stdio.h>
 #include <stdlib.h>
 
-#include "string.h"
+#include "string_lib.h"
 
-char* assert_int(int a, int b);
+char* assert_int(int, int);
+char* assert_true(int);
+char* assert_false(int);
 
-int main(void) {
-  printf("string test =============\n");
+void run_tests(void) {
+  printf(" ============= string test =============\n");
+
+  printf("-> str_len:\n");
+  printf("- assert base length %s\n", assert_int(str_len("yo"), 2));
+  printf("- assert empty string %s\n", assert_int(str_len(""), 0));
+  printf("- assert null string %s\n", assert_int(str_len(NULL), 0));
 
-  printf("strlen:\n");
-  printf("- assert base length %s\n", assert_int(strlen("yo"), 2));
-  printf("- assert empty string %s\n", assert_int(strlen(""), 0));
-  printf("- assert null string %s\n", assert_int(strlen(NULL), 2));
+  printf("-> str_cmp:\n");
+  printf("- assert equal string %s\n", assert_int(str_cmp("yo", "yo"), 0));
+  printf("- assert equal string %s\n", assert_int(str_cmp("yoo", "yo"), 1));
+  printf("- assert equal string %s\n", assert_int(str_cmp("yooo", "yo"), 2));
+  printf("- assert equal string %s\n", assert_int(str_cmp("yoooo", "yo"), 3));
+  printf("- assert equal string %s\n", assert_int(str_cmp("yo", "yoo"), -1));
+  printf("- assert equal string %s\n", assert_int(str_cmp("yo", "yooo"), -2));
+}
```

Comment:
```suggestion
  printf("- assert equal string %s\n", assert_int(str_cmp("yo", "yo"), 0));
  printf("- assert equal string %s\n", assert_int(str_cmp("yoo", "yo"), 1));
  printf("- assert equal string %s\n", assert_int(str_cmp("yooo", "yo"), 1));
  printf("- assert equal string %s\n", assert_int(str_cmp("yoooo", "yo"), 1));
  printf("- assert equal string %s\n", assert_int(str_cmp("yo", "yoo"), -1));
  printf("- assert equal string %s\n", assert_int(str_cmp("yo", "yooo"), -1));
  printf("- assert equal string %s\n", assert_int(str_cmp("abc", "bcd"), -1));
  printf("- assert equal string %s\n", assert_int(str_cmp("bcd", "abc"), 1));
}
```
The last 2 are cases that would be `0` with your current code.